src/stactools/sentinel1/grd/properties.py

```diff
@@ -59,12 +72,16 @@ def fill_sar_properties(sar_ext: SarExtension, manifest: XmlElement,
     sar_ext.observation_direction = ObservationDirection.RIGHT
 
     # Read properties
-    sar_ext.instrument_mode = manifest.findall(".//s1sarl1:mode")[0].text
+    instrument_mode = manifest.find_text(".//s1sarl1:mode")
+    if instrument_mode:
+        sar_ext.instrument_mode = str(instrument_mode)
```

Comment:
If `instrument_mode` (is not `None`), then it is automatically narrowed to type `str`, so using the `str` function is unnecessary for the type checker to be satisified.

```suggestion
        sar_ext.instrument_mode = instrument_mode
```